api/v1alpha1/linodecluster_types.go

```diff
@@ -35,6 +36,10 @@ type LinodeClusterSpec struct {
 	// NetworkSpec encapsulates all things related to Linode network.
 	// +optional
 	Network NetworkSpec `json:"network"`
+
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Value is immutable"
+	// +optional
+	VPCRef *corev1.ObjectReference `json:"vpcRef,omitempty"`
```

Comment:
Do we plan to support CAPI creating VPCs for your cluster if not provided? if so we might want to maybe add a VPC section or something like that to support some options that might be used other than an external VPC ref

Reply:
@eljohnson92 I can imagine a VPCTemplate similar to Machine Template. But i think we should address it in a new PR. At the moment VPC API is not available, so the only option is bring your own vpc at the moment.

Reply:
It looks like there are client packages for VPCs [here](https://github.com/linode/linodego/blob/main/vpc.go) do we need to just use the beta API with the client in order to use it?

Reply:
@eljohnson92 I'm not pretty sure. I couldn't create VPC via CLI or API on v4. Web UI uses v4beta as version, but this version is not available with my personal token. Documentation also mentions this story.

Reply:
@mhmxs in the linodego [client](https://github.com/linode/linodego/blob/main/client.go#L57) we can set apiVersion to `v4beta` to access the beta api. I would be fine tackling this in a later PR but I just want to make sure we account for those changes in this interface

Reply:
@eljohnson92 Your points are valid, just should we continue discussion at the ticket? https://github.com/linode/cluster-api-provider-linode/issues/4